### PR TITLE
UX: Use dvh for sidebar height

### PR DESCRIPTION
**Pull request by davidtaylorhq**
This will automatically adjust when browser UI is shown/hidden (e.g. when scrolling up/down on mobile Safari).

Similar approach to https://github.com/discourse/discourse/commit/c82094cd9dbd6160a2e611e6dde915ce4695b32f, which targeted the 'slide-in' version of menus.

### CODE REVIEW

**Review by CvX — APPROVED** [review]

File: app/assets/stylesheets/common/base/sidebar.scss [hunk 1]
```diff
@@ -31,13 +31,19 @@
...
+  --100dvh: 100vh;
+  @supports (height: 100dvh) {
+    --100dvh: 100dvh;
+  }
+
   .footer-nav-ipad & {
     top: calc(var(--header-offset) + var(--footer-nav-height));
     height: calc(
-      100vh - var(--header-offset, 0px) - var(--footer-nav-height, 0px)
+      var(--100dvh) - var(--header-offset, 0px) - var(--footer-nav-height, 0px)
```
Comment (pmusaraj) [inline app/assets/stylesheets/common/base/sidebar.scss:43]: Any reason not to simply use `100dvh` for the special iPad case and not  introduce a new custom property? As far as I can tell, this is very much scoped to Safari on iPads.

File: app/assets/stylesheets/common/base/sidebar.scss [hunk 2]
```diff
@@ -31,13 +31,19 @@
...
     );
   }
-  height: calc(100vh - var(--header-offset, 0px));
+  height: calc(var(--100dvh) - var(--header-offset, 0px));
```
Comment (pmusaraj) [inline app/assets/stylesheets/common/base/sidebar.scss:46]: Not new to your PR, but I don't like seeing the general height declaration added _after_ the specific `.footer-nav-ipad & {` one. I would move this above just to be extra sure about the order of things.

File: app/assets/stylesheets/common/base/sidebar.scss (hunk 1 above)
Comment (davidtaylorhq) [reply app/assets/stylesheets/common/base/sidebar.scss:43]: So you're saying we should only use `dvh` for the `.footer-nav-ipad` `height`? And go back to `vh` for the regular `height` entry?

`.footer-nav-ipad` only applies to the PWA view on iPad. The motivation for this PR is problems with regular Safari on iPad. So think we need to use `dvh` in both places.

File: app/assets/stylesheets/common/base/sidebar.scss (hunk 2 above)
Comment (davidtaylorhq) [reply app/assets/stylesheets/common/base/sidebar.scss:46]: CSS will always prefer the 'most specific' selector, regardless of order. But totally happy to re-arrange if you think it makes it easier to read 👌

Comment (pmusaraj) [reply app/assets/stylesheets/common/base/sidebar.scss]: Thanks!

File: app/assets/stylesheets/common/base/sidebar.scss (hunk 1 above)
Comment (pmusaraj) [reply app/assets/stylesheets/common/base/sidebar.scss:45]: Ah, that makes sense.

**Pull request merged**